app/notification.go

```diff
@@ -403,6 +410,21 @@ func (a *App) SendNotifications(post *model.Post, team *model.Team, channel *mod
 	}
 
 	a.Publish(message)
+	// If this is a reply in a thread, notify participants
+	if *a.Config().ServiceSettings.ThreadAutoFollow && post.RootId != "" {
```

Comment:
Why only if `ThreadAutoFollow` is true? If this is false, and a user is following the thread, then will they not get a notification? 
And conversely, a user who participated in the thread but then unfollowed the thread will be notified?

Reply:
You are absolutely right @ashishbhate . I forgot to introduce the new CollapsedThreads config setting.